server/src/main/java/org/opensearch/cluster/store/RemoteClusterStateService.java

```diff
@@ -0,0 +1,247 @@
+            final ClusterMetadataMarker marker = new ClusterMetadataMarker(
+                uploadedIndexMetadata,
+                clusterState.term(),
+                clusterState.getVersion(),
+                clusterState.metadata().clusterUUID(),
+                clusterState.stateUUID()
+            );
+            writeMetadataMarker(clusterState.getClusterName().value(), clusterState.metadata().clusterUUID(), marker, markerFileName);
+            return marker;
+        }
+    }
+
+    private String writeIndexMetadata(String clusterName, String clusterUUID, IndexMetadata indexMetadata, String fileName)
+        throws IOException {
+        final BlobContainer indexMetadataContainer = indexMetadataContainer(clusterName, clusterUUID, indexMetadata.getIndexUUID());
+        INDEX_METADATA_FORMAT.write(indexMetadata, indexMetadataContainer, fileName, blobStoreRepository.getCompressor());
+        // returning full path
+        return indexMetadataContainer.path().buildAsString() + fileName;
+    }
+
+    private void writeMetadataMarker(String clusterName, String clusterUUID, ClusterMetadataMarker marker, String fileName)
+        throws IOException {
+        final BlobContainer metadataMarkerContainer = markerContainer(clusterName, clusterUUID);
+        CLUSTER_METADATA_MARKER_FORMAT.write(marker, metadataMarkerContainer, fileName, blobStoreRepository.getCompressor());
+    }
+
+    private BlobContainer indexMetadataContainer(String clusterName, String clusterUUID, String indexUUID) {
+        // 123456789012_test-cluster/cluster-state/dsgYj10Nkso7/index/ftqsCnn9TgOX
+        return blobStoreRepository.blobStore()
+            .blobContainer(
+                blobStoreRepository.basePath()
+                    .add(Base64.getUrlEncoder().withoutPadding().encodeToString(clusterName.getBytes(StandardCharsets.UTF_8)))
+                    .add("cluster-state")
+                    .add(clusterUUID)
+                    .add("index")
+                    .add(indexUUID)
+            );
+    }
+
+    private BlobContainer markerContainer(String clusterName, String clusterUUID) {
+        // 123456789012_test-cluster/cluster-state/dsgYj10Nkso7/marker
+        return blobStoreRepository.blobStore()
+            .blobContainer(
+                blobStoreRepository.basePath()
+                    .add(Base64.getUrlEncoder().withoutPadding().encodeToString(clusterName.getBytes(StandardCharsets.UTF_8)))
+                    .add("cluster-state")
+                    .add(clusterUUID)
+                    .add("marker")
+            );
+    }
+
+    private static String getMarkerFileName(long term, long version) {
+        // 123456789012_test-cluster/cluster-state/dsgYj10Nkso7/marker/2147483642_2147483637_456536447_marker
+        return String.join(
+            DELIMITER,
+            "marker",
+            String.valueOf(Long.MAX_VALUE - term),
```

Comment:
Is this used to fetch latest file based on lexicographic sorting? If yes, we need to use padding to have equal length strings. If we use unequal length strings, `12345` is considered lower than `978`. 

Check `RemoteStoreUtils.invertLong`